src/tools/UpdatePackageDependencies.ts

```diff
@@ -6,33 +6,47 @@
 import * as fs from 'fs';
 import * as os from 'os';
 import { Package } from '../packageManager/Package';
-
+import { DownloadFile } from '../packageManager/FileDownloader';
+import { EventStream } from '../EventStream';
+import * as Event from "../omnisharp/loggingEvents";
+import NetworkSettings, { NetworkSettingsProvider } from '../NetworkSettings';
+import { getBufferIntegrityHash } from '../packageManager/isValidDownload';
 
 interface PackageJSONFile
 {
     runtimeDependencies : Package[];
 }
 
-export function updatePackageDependencies() {
+export async function updatePackageDependencies() : Promise<void> {
 
-    const urlsIndex = process.argv.indexOf("--urls");
-    const newPrimaryUrls = urlsIndex >= 0 ? process.argv[urlsIndex+1] : undefined;
-    
-    const fallbackUrlsIndex = process.argv.indexOf("--fallbackUrls");
-    const newFallbackUrls = fallbackUrlsIndex >= 0 ? process.argv[fallbackUrlsIndex+1] : undefined;
+    const newPrimaryUrls = process.env["NEW_DEPS_URLS"];
+    const newVersion = process.env["NEW_DEPS_VERSION"];
     
-    if (newPrimaryUrls === undefined || newPrimaryUrls === "-?" || newPrimaryUrls === "-h") {
-        console.log("This command will update the URLs for package dependencies in package.json");
+    if (!newPrimaryUrls || !newVersion) {
+        console.log();
+        console.log("'npm run gulp updatePackageDependencies' will update package.json with new URLs of dependencies.");
         console.log();
-        console.log("Syntax: updatePackageDependencies --urls \"<url1>,<url2>,...\" [--fallbackUrls \"<fallback-url-1>,<fallback-url-2>...\"]");
+        console.log("To use:");
+        const setEnvVarPrefix = os.platform() === 'win32' ? "set " : "export ";
+        const setEnvVarQuote = os.platform() === 'win32' ? "" : "\'";
+        console.log(`  ${setEnvVarPrefix}NEW_DEPS_URLS=${setEnvVarQuote}https://example1/foo-osx.zip,https://example1/foo-win.zip,https://example1/foo-linux.zip${setEnvVarQuote}`);
+        console.log(`  ${setEnvVarPrefix}NEW_DEPS_VERSION=${setEnvVarQuote}1.2.3${setEnvVarQuote}`);
+        console.log("  npm run gulp updatePackageDependencies");
         console.log();
         return;
     }
 
-    if (newPrimaryUrls.length === 0) {
-        throw new Error("Invalid first argument to updatePackageDependencies. URL string argument expected.");
+    const newPrimaryUrlArray = newPrimaryUrls.split(',');
+    for (let urlToUpdate of newPrimaryUrlArray) {
+        if (!urlToUpdate.startsWith("https://")) {
+            throw new Error("Unexpected 'NEW_DEPS_URLS' value. All URLs should start with 'https://'.");
+        }
     }
 
+    if (! /^[0-9]+\.[0-9]+\.[0-9]+$/.test(newVersion)) {
```

Comment:
We could also do a semver check here

Reply:
I don't think there is anything in 'semver' that supports searching for a semver in a string, and since we always want the input version to be subsequently searchable, I don't think it makes sense to use the semver library for this.